lib/kwtool/crawlers/schemas/keyword.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Kwtool.Crawlers.Schemas.Keyword do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  @statuses %{
+    added: "added",
+    in_process: "in_process",
+    finished: "finished",
+    done: "done"
+  }
+
+  schema "keywords" do
+    field :phrase, :string
+    field :raw_result, :string
+    field :status, :string
+
+    timestamps()
+
+    belongs_to :user, Kwtool.Accounts.Schemas.User
+  end
+
+  def changeset(keyword \\ %__MODULE__{}, attrs) do
+    keyword
+    |> cast(attrs, [:phrase, :raw_result, :status, :user_id])
+    |> set_status()
+    |> validate_required([:phrase, :status, :user_id])
+    |> validate_inclusion(:status, Map.values(@statuses))
+    |> assoc_constraint(:user)
+  end
+
+  defp set_status(%Ecto.Changeset{data: %Kwtool.Crawlers.Schemas.Keyword{status: nil}} = changeset) do
```

Comment:
I'm not so sure about this one. is it going to raise exception if status is not nil 🤔

Reply:
I added another method to match the case `status` is **not** `nill`👍

Reply:
Would it fail in case when the record already has the status set.

e.g. `Keyword.changeset(%Keyword{status: "added"}, %{status: "in_process"})`

I think maybe it's safer if we have separated changeset
1. For create, we always set the status to `added`
2. For update, we don't set any status

For example, maybe something like this

```elixir
def create_changeset(keyword \\ %__MODULE__{}, attrs) do
   keyword
   |> put_change(:status, "added")
   |> changeset(attrs)
end

def update_changeset(keyword \\ %__MODULE__{}, attrs) do
   changeset(keyword, attrs)
end

defp changeset(keyword \\ %__MODULE__{}, attrs) do
    keyword
    |> cast(attrs, [:phrase, :raw_result, :status, :user_id])
    |> validate_required([:phrase, :status, :user_id])
    |> validate_inclusion(:status, Map.values(@statuses))
    |> assoc_constraint(:user)
end
```

Reply:
or if we only update the status we can have 😆 

```elixir 
def create_changeset(keyword \\ %__MODULE__{}, attrs) do
   keyword
   |> cast(attrs, [:phrase, :raw_result, :status, :user_id])
   |> put_change(:status, "added")
   |> validate_required([:phrase, :status, :user_id])
   |> validate_inclusion(:status, Map.values(@statuses))
   |> assoc_constraint(:user)
end

def update_changeset(keyword \\ %__MODULE__{}, attrs) do
   keyword
   |> cast(attrs, [:raw_result, :status])
   |> validate_inclusion(:status, Map.values(@statuses))
end
```

Reply:
yes, now i changed it to `create_changeset` instead.